From aed67387933a934e3734b9768a9467294569952b Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Sat, 14 Dec 2024 09:12:49 +0000
Subject: [PATCH 4/4] Use `futimens` instead of legacy `futimes`

wasi-libc does not provide `futimes` as it is a legacy function.
https://github.com/WebAssembly/wasi-libc/blob/574b88da481569b65a237cb80daf9a2d5aeaf82d/libc-top-half/musl/include/sys/time.h#L34
---
 .../FoundationEssentials/FileManager/FileOperations.swift   | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Sources/FoundationEssentials/FileManager/FileOperations.swift b/Sources/FoundationEssentials/FileManager/FileOperations.swift
index 49e2c37..2ef8a03 100644
--- a/Sources/FoundationEssentials/FileManager/FileOperations.swift
+++ b/Sources/FoundationEssentials/FileManager/FileOperations.swift
@@ -988,11 +988,11 @@ enum _FileOperations {
             #endif
             
             // Copy modification date
-            let value = timeval(tv_sec: statInfo.st_mtim.tv_sec, tv_usec: statInfo.st_mtim.tv_nsec / 1000)
+            let value = statInfo.st_mtim
             var tv = (value, value)
             try withUnsafePointer(to: &tv) {
-                try $0.withMemoryRebound(to: timeval.self, capacity: 2) {
-                    if futimes(dstFD, $0) != 0 {
+                try $0.withMemoryRebound(to: timespec.self, capacity: 2) {
+                    if futimens(dstFD, $0) != 0 {
                         try delegate.throwIfNecessary(errno, srcPath(), dstPath())
                     }
                 }
-- 
2.46.0

